polkadot/primitives/src/vstaging/mod.rs

```diff
@@ -586,55 +594,70 @@ impl<H: Copy> CommittedCandidateReceiptV2<H> {
 	/// Checks if descriptor core index is equal to the committed core index.
 	/// Input `cores_per_para` is a claim queue snapshot stored as a mapping
 	/// between `ParaId` and the cores assigned per depth.
+	/// `core_index_enabled` optionally describes the status of the elastic scaling MVP node
+	/// feature.
 	pub fn check_core_index(
 		&self,
 		cores_per_para: &TransposedClaimQueue,
-	) -> Result<(), CandidateReceiptError> {
+		core_index_enabled: Option<bool>,
```

Comment:
By assuming we always have the core index injection enabled we don't need to add this new argument. It very easy to make this assumption since it is also a bug fix.